custom_components/hacs/utils/queue_manager.py

```diff
@@ -50,32 +51,38 @@ async def execute(self, number_of_tasks: int | None = None) -> None:
 
         self.running = True
 
-        _LOGGER.debug("<QueueManager> Checking out tasks to execute")
-        local_queue = []
-
-        if number_of_tasks:
-            for task in self.queue[:number_of_tasks]:
-                local_queue.append(task)
-        else:
-            for task in self.queue:
-                local_queue.append(task)
-
-        for task in local_queue:
-            self.queue.remove(task)
-
-        _LOGGER.debug("<QueueManager> Starting queue execution for %s tasks", len(local_queue))
-        start = time.time()
-        result = await asyncio.gather(*local_queue, return_exceptions=True)
-        for entry in result:
-            if isinstance(entry, Exception):
-                _LOGGER.error("<QueueManager> %s", entry)
-        end = time.time() - start
-
-        _LOGGER.debug(
-            "<QueueManager> Queue execution finished for %s tasks finished in %.2f seconds",
-            len(local_queue),
-            end,
-        )
+        while self.queue and (number_of_tasks is None or number_of_tasks):
+            _LOGGER.debug("<QueueManager> Checking out tasks to execute")
+            local_queue = []
+            queue_item = self.queue[0]
+
+            if number_of_tasks:
+                for task in queue_item[:number_of_tasks]:
+                    local_queue.append(task)
+                    number_of_tasks -= 1
+            else:
+                for task in queue_item:
+                    local_queue.append(task)
+
+            for task in local_queue:
```

Comment:
Looking now, this looks like a bug?
These should be removed after the items have been executed right?
So the properties exposed in this class would be correct.